railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -91,7 +92,7 @@ def ensure_credentials_have_been_added
 
         def change_credentials_in_system_editor
           credentials.change do |tmp_path|
-            system("#{ENV["EDITOR"]} #{tmp_path}")
+            system("#{ENV["EDITOR"]} #{Shellwords.escape(tmp_path)}")
```

Comment:
Apologies for dogpiling :dog:, but Ruby provides a [`system`](https://docs.ruby-lang.org/en/2.7.0/Kernel.html#method-i-system) overload that can be helpful here:

```suggestion
            system(ENV["EDITOR"], tmp_path)
```

Reply:
Oh taht's cooool